moveit_ros/moveit_servo/test/basic_servo_tests.cpp

```diff
@@ -170,6 +174,71 @@ TEST_F(ServoFixture, SendJointServoTest)
   EXPECT_LT(received_count, num_commands + 20);
   servo_->setPaused(true);
 }
+
+// This a friend test of a private member function
+TEST_F(ServoFixture, EnforceVelLimitsTest)
+{
+  auto parameters = servo_->getParameters();
+  const double publish_period = parameters.publish_period;
+
+  // Request joint angle changes that are too fast, given the control period in servo settings YAML file.
+  Eigen::ArrayXd delta_theta(7);
+  delta_theta[0] = 0;  // rad
+  delta_theta[1] = 0.01;
+  delta_theta[2] = 0.02;
+  delta_theta[3] = 0.03;
+  delta_theta[4] = 0.04;
+  delta_theta[5] = 0.05;
+  delta_theta[6] = 0.06;
+
+  // Store the original joint commands for comparison before applying velocity scaling.
+  Eigen::ArrayXd orig_delta_theta = delta_theta;
+  enforceVelLimits(delta_theta);
+
+  // From Panda arm MoveIt joint_limits.yaml. The largest velocity limits for a joint.
+  const double panda_max_joint_vel = 2.610;  // rad/s
```

Comment:
Since `joint_limits.yaml` take precedence over the URDF the value is actually `2.61` and not `2.871`. Took me a while to understand that that was the source of errors. :)

Reply:
good catch there